src/coreclr/jit/helperexpansion.cpp

```diff
@@ -868,7 +868,14 @@ bool Compiler::fgExpandHelperForBlock(BasicBlock** pBlock)
 
         for (GenTree* const tree : stmt->TreeList())
         {
-            if (!tree->IsHelperCall())
+            if (handleIntrinsics)
+            {
+                if (!tree->IsCall() || ((tree->AsCall()->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC) == 0))
+                {
+                    continue;
+                }
+            }
+            else if (!tree->IsHelperCall())
```

Comment:
Seems like it would be cleaner to leave this up to the expansion function and avoid the `handleIntrinsics` argument. The expansion function presumably has to do a bunch of call related checks anyway, so it can check for `IsHelperCall` or special intrinsic as necessary.